hudi-common/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadata.java

```diff
@@ -217,6 +214,12 @@ private void openReadersIfNeeded() throws IOException {
           .sorted(HoodieLogFile.getLogFileComparator())
           .map(o -> o.getPath().toString())
           .collect(Collectors.toList());
+
+      // Only those log files which have a corresponding completed instant on the dataset should be read
+      // This is because the metadata table is updated before the dataset instants are committed.
+      Set<String> validInstantTimestamps = datasetMetaClient.getActiveTimeline().filterCompletedInstants().getInstants()
+          .map(i -> i.getTimestamp()).collect(Collectors.toSet());
+
       Option<HoodieInstant> lastInstant = metaClient.getActiveTimeline().filterCompletedInstants().lastInstant();
       String latestMetaInstantTimestamp = lastInstant.map(HoodieInstant::getTimestamp).orElse(SOLO_COMMIT_TIMESTAMP);
```

Comment:
latestMetaInstantTimestamp is same as the last instant on completedInstants of the dataset, isn't?